types/next-auth/client.d.ts

```diff
@@ -28,6 +28,19 @@ interface SetOptionsParams {
     keepAlive?: number;
 }
 
+interface SignInObjectReturn {
+    error: string | undefined;
+    status: number;
+    ok: boolean;
+    url: string | null;
+}
+
+type SignInDataParams = GenericObject & {
+  callbackUrl?: string;
```

Comment:
You forgot the `redirect?: false` option. It defaults to true.
https://next-auth.js.org/getting-started/client#using-the-redirect-false-option

In theory, we could add a generic here, because `redirect` only makes sense, if you set the first param of `signIn` to either `email` or `credentials`.

Also, why not inline as the signout params? Don't have a problem with either, we should just be consistent